include/jemalloc/internal/extent.h

```diff
@@ -333,12 +342,20 @@ extent_slab_data_get_const(const extent_t *extent) {
 	return &extent->e_slab_data;
 }
 
-static inline void
-extent_prof_info_get(const extent_t *extent, prof_info_t *prof_info) {
-	assert(prof_info != NULL);
-	prof_info->alloc_tctx = (prof_tctx_t *)atomic_load_p(
-	    &extent->e_prof_tctx, ATOMIC_ACQUIRE);
-	prof_info->alloc_time = extent->e_alloc_time;
+static inline prof_tctx_t *
+extent_prof_tctx_get(const extent_t *extent) {
+	return (prof_tctx_t *)atomic_load_p(&extent->e_prof_info.e_prof_tctx,
+	    ATOMIC_ACQUIRE);
+}
+
+static inline const nstime_t *
+extent_prof_alloc_time_get(const extent_t *extent) {
+	return &extent->e_prof_info.e_prof_alloc_time;
+}
+
+static inline prof_recent_t *
+extent_prof_recent_alloc_get(const extent_t *extent) {
```

Comment:
This would make `extent.h` dependent on `mutex.h` as well as `prof_recent_alloc_mtx` which is internal to `prof_recent.c`. How about defining these functions inside `prof_recent.c`, so as to make the code structure less interweaving?

Reply:
Both sound fine to me. I’ll leave it up to you.